Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -436,6 +438,21 @@ Bytecode::CodeGenerationErrorOr<Optional<ScopedOperand>> RegExpLiteral::generate
     return dst;
 }
 
+static bool can_emit_global_lookup_for_current_lexical_environment(VM& vm)
+{
+    // If the current execution context has overridden the lexical environment with a `with` object environment, we
+    // cannot optimize identifier lookup via global access.
+    if (vm.execution_context_stack().is_empty())
+        return true;
+
+    auto lexical_environment = vm.running_execution_context().lexical_environment;
+    if (!lexical_environment || !is<ObjectEnvironment>(*lexical_environment))
+        return true;
+
+    auto const& object_environment = static_cast<ObjectEnvironment const&>(*lexical_environment);
+    return !object_environment.is_with_environment();
+}
```

Comment:
Until now, we have not looked at current VM state while generating bytecode, so this introduces an awkward new thing where codegen depends on the current execution stack.

Is there some way we can do this without introducing that kind of sneaky relationship?

Reply:
I haven't spent a ton of time on this, but wasn't able to come up with an alternate fix yet, so I've pushed just reverting the previous broken fix for now until we figure something else out